Style person page

johnbxu opened the pull request:
- Styled person page
- Added django debug toolbar for development

gabrieltanchen commented on cigionline/static/css/_people.scss:
```diff
@@ -0,0 +1,107 @@
+    }
+
+    .hero-main {
+      max-width: 33rem;
```
This isn't being applied.

gabrieltanchen commented on cigionline/static/css/_people.scss:
```diff
@@ -0,0 +1,107 @@
+      padding: 1em;
+
+      .expertise-item {
+        &::after {
```
We use this in a number of places in our site - makes sense for it to be a mixin.

gabrieltanchen commented on cigionline/static/css/_people.scss:
```diff
@@ -0,0 +1,107 @@
+    }
+
+    .full-bio {
+      margin: 3em 0;
```
Rather than individually setting margins here - use the `section` tag in the template and set the margin for `section`s within an `article`.

gabrieltanchen commented on people/tests.py:
```diff
@@ -624,3 +626,12 @@ def test_personpage_child_page_types(self):
         )
+
+    def test_personpage_is_staff_should_return_correct_boolean(self):
+        staff = PersonPage.objects.get(title="Staff Live")
```
Add all the person types here please.

gabrieltanchen commented on research/tests.py:
```diff
@@ -77,28 +77,62 @@ def test_topicpage_child_page_types(self):
+        rendered = self.TEMPLATE.render(Context({'test_topics': test_topics}))
+        self.assertNotIn('topic1', rendered)
 
     def test_topics_archived_do_not_render(self):
```
Archived topics should render.

gabrieltanchen commented on templates/people/person_page.html:
```diff
@@ -1,5 +1,201 @@
+
+<section class="main container-sm">
+  {% if not page.archive %}
+    {% if page.is_staff == False %}
```
The expertise section does show up for staff - take Aaron Shull for example.

gabrieltanchen commented on templates/people/person_page.html:
```diff
@@ -1,5 +1,201 @@
+  </section>
+{% endif %}
+
+<section class="main container-sm">
```
Don't use a big container here. Each of the sections should be their own sections with their own container classes. In this case the container classes are all the same, but that won't be the case for opinions - or especially, for opinion series.

gabrieltanchen commented on templates/people/person_page.html:
```diff
@@ -1,5 +1,201 @@
+    {% with publication.value.publication_type|lower as type %}
+      <div class="external-publications">
+        <h2>Select Publications</h2>
+        <ol>
```
Rather than using a list, each of these should be `article` tags. For reading: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article

johnbxu commented on research/templatetags/topic_tags.py:
```diff
@@ -8,3 +8,9 @@
+
+@register.inclusion_tag('research/topics.html')
+def topics(topics):
+    topics = topics.live().filter(archive=0).order_by('title')
```
Sara Seck is archived. On live, the topics don't show up for non-archived people. I'll ask Som if this behaviour is intended.

johnbxu replied:
Som says to hide topics for both archived and unarchived experts.

johnbxu commented:
Made these changes:
- refactored ol, ul styling and margins to `article { section {...} }`
- added .paragraph-heading class 
- added `comma-separated` mixin
- replaced li with article in external publications
- removed is_staff tag
- added topics property in PersonPage model
- restructured person_page template to wrap everything in an `article`, which contains `hero` and `main` sections. In main section, there are multiple child sections. Not sure if this makes sense?

Ready for review.

gabrieltanchen commented on templates/people/person_page.html:
```diff
@@ -1,5 +1,190 @@
 
 {% block content %}
-  {% include "includes/hero_standard.html" with title=self.title %}
+{% with page.expertise as expertises %}
```
Don't stack `with` statements like this, move each one to its relevant bit of code.

gabrieltanchen commented on templates/people/person_page.html:
```diff
@@ -1,5 +1,190 @@
+  </section>
+
+  {% if not page.archive %}
+    <section class="main">
```
This main section is redundant, it's fine to remove this and let any content sections flow after the hero section.

johnbxu commented:
I ended up just removing the `with`s since they don't really add that much readability. Also removed the main section. Ready for review.

gabrieltanchen commented on templates/people/person_page.html:
```diff
@@ -1,5 +1,179 @@
+            {% if page.twitter_username %}
+              <div class="contact-item">
+                <i class="fab fa-twitter" aria-hidden="true"></i>
+                <span><a href="https://twitter.com/{{ twitter }}" target="_blank">{{ page.twitter_username }}</a></span>
```
URL here needs to be updated after removing the `with`.

johnbxu commented:
Updated variable name. Ready for review.
